### ISSUE TITLE
Chef::Exceptions::ContentLengthMismatch: Response body length

### ISSUE THREAD

**Issue by EmFl** [higanworks/knife-zero#69]
Hello,

I'm trying to use knife zero but failing to upload the cookbooks to remote machine with the following error : 
<code>
Synchronizing Cookbooks:
- apache2 (3.1.0)
- xml (1.3.1)
- build-essential (2.2.4)
- mysql (6.1.2)
- yum-epel (0.6.4)
- windows (1.38.3)
- iis (4.1.4)
- chef-sugar (3.1.1)
- yum-mysql-community (0.1.19)
- smf (2.2.7)
- yum (3.8.2)
- chef_handler (1.2.0)
- rbac (1.0.3)

================================================================================
Error Syncing Cookbooks:
================================================================================

Unexpected Error:
 -----------------
Chef::Exceptions::ContentLengthMismatch: Response body length 3383108 does not match HTTP Content-Length header 3658571.
</code>

Any idea what could be wrong with my setup ?
Version of chef : <code>Starting Chef Client, version 12.5.1</code>

**Comment by EmFl** [comment]
Deleting /var/chef/cache on the remote machine solved the issue.

**Comment by sawanoboly** [comment]
Thank you for your report :smiley: 

I'll append this workaround to [Document](http://knife-zero.github.io) later.

**Comment by umegaya** [comment]
hi, I encounter same issue. I try to use knife zero via bastion ssh (that is, my host =(ssh)> admin server =(ssh)> target server). after I copy my chef repo to admin server and run knife zero on there, then knife zero start to run correctly. so I suspect there is some issue about running knife zero under bastion ssh environment. 

here is my .ssh/config (detail is not same). I hope this would helps.
```
host gateway
   HostName adminserver
   User admin

Host targetserver
  User admin
  ProxyCommand ssh -W %h:%p adminserver
```

**Comment by sawanoboly** [comment]
@umegaya Could you tell me a distribution of gateway server and version of sshd?

I've never seen this problem while using ubuntu or SmartOS as gateway... :sweat:

**Comment by umegaya** [comment]
@sawanoboly 
our server is CentOS 6.7 (64bit) and sshd is installed by openssh-server-5.3p1-112.el6_7.rpm. 

anyway, knife zero works great and we appreciate your fast response. keep up good work :D

**Comment by sawanoboly** [comment]
hmm.. I'll create CentOS VM to try reproduce this problem.
I perhaps find bug or implement workaround.  But, don’t expect too much.

That aside, could you try another gateway which has newer open-ssh server than 5.3 ?

**Comment by umegaya** [comment]
sorry, this is only gateway can be used, and difficult to change sshd version without permission. we will try to do so, but it may take some time to be granted (or may rejected). 

when we update sshd, we will report whether problem is gone or not. thanks!

**Comment by sawanoboly** [comment]
I've noticed that Webrick Server (It's used by Chef-Zero as middleware.) offten give up transferring binary file.

For example,

- the file is large.
- when the transfer speed is slow.

To recover this, we have to clear caches and retry. Since it may be modified in the patch to Chef-Zero, I'll take a look.

**Comment by sawanoboly** [comment]
Good news :smile:  @EmFl @umegaya 

This problem probably will be fixed in next net-ssh 2.9.x release.

https://github.com/net-ssh/net-ssh/pull/280

If we want to use it as soon as possible, we should add line to Gemfile like below.

```
gem 'net-ssh', github: 'net-ssh/net-ssh', ref: '2-9-release'
```

**Comment by umegaya** [comment]
@sawanoboly hi, today this problem reproduced, so I try your work around, problem seems to be fixed. thank you for your advice :D
and sorry for my laziness, but this problem did not happened in my environment for a while.

**Comment by sawanoboly** [comment]
Thank you for report :+1: 

This issue will be kept open until release net-ssh 2.9.3 as reminder for me.

**Comment by sawanoboly** [comment]
resolved with net-ssh 2.9.4.
